modules/targetanalysis/include/pixel_object_list.h

```diff
@@ -0,0 +1,237 @@
+/**
+ * @file pixel_object_list.h
+ * @author WARG
+ *
+ * @section LICENSE
+ *
+ *  Copyright (c) 2015-2017, Waterloo Aerial Robotics Group (WARG)
+ *  All rights reserved.
+ *
+ *  This software is licensed under a modified version of the BSD 3 clause license
+ *  that should have been included with this software in a file called COPYING.txt
+ *  Otherwise it is available at:
+ *  https://raw.githubusercontent.com/UWARG/computer-vision/master/COPYING.txt
+ */
+
+
+#ifndef PIXEL_OBJECT_LIST_H_INCLUDED
+#define PIXEL_OBJECT_LIST_H_INCLUDED
+
+#include "frame.h"
+#include "object.h"
+#include "pixel_object.h"
+#include <vector>
+
+//Macros and Definitions
+#define RAD2DEG(rad) ((rad)*180.0/M_PI)
+#define DEG2RAD(deg) ((deg)*M_PI/180.0)
+#define EARTH_RADIUS 6371000
+
+/**
+ * @class PixelObjectList
+ *  
+ * PixelObjectList describes a structure for storing pixeltargets and grouping
+ * similar ones together. The duplicate and unique forms can then be later found
+ * within a unique 'Object' class. The object class contains all pixel objects
+ * as well as their averaged data, with accompanying error rates.
+ *
+ * Module geolocates targets using their pixel locations, visual contour
+ * and photo metadata, determines target type and calculates 
+ * possible error. As targets are processed unique targets will 
+ * be identified and the data combined into a single object.
+ *
+ * This class is a singleton.
```

Comment:
Agreed. The slow part is still the TargetId stuff, so a single thread should be okay...
You are right multiple threads in list comparisons could get a little bit tricky.

IE: Object 1 and object 2 are the same, but due to concurrency don't ever get properly compared and are both appended to the list as a unique object.